Content.Client/HealthAnalyzer/UI/HealthAnalyzerWindow.xaml.cs

```diff
@@ -57,15 +57,20 @@ public void Populate(HealthAnalyzerScannedUserMessage msg)
                 // Show the total damage and type breakdown for each damage group.
                 foreach (var (damageGroupId, damageAmount) in damagePerGroup)
                 {
+                    if (damageAmount == 0)
+                    {
+                        continue;
+                    }
                     text.Append($"\n{Loc.GetString("health-analyzer-window-damage-group-text", ("damageGroup", Loc.GetString("health-analyzer-window-damage-group-" + damageGroupId)), ("amount", damageAmount))}");
+
                     // Show the damage for each type in that group.
                     var group = protos.Index<DamageGroupPrototype>(damageGroupId);
                     foreach (var type in group.DamageTypes)
                     {
-                        if (damagePerType.TryGetValue(type, out var typeAmount))
+                        if (damagePerType.TryGetValue(type, out var typeAmount) && damageAmount != 0)
```

Comment:
is checking `damageAmount != 0` necessary? You already checked it on line 60.

Reply:
I think it was because damage and damage groups are printed seperately and both needed to be checked

Reply:
It doesn't look like there are any assignments to `damageAmount` in the loop, and since on line 60 you skip if `damageAmount == 0`, then on line 70 `damageAmount` cannot be equal to zero, therefore this is a dead comparison.

Reply:
he's correct. just remove it.

Reply:
Haven't opened git in a few days sor

Reply:
![image](https://github.com/space-wizards/space-station-14/assets/121047731/e8b846c0-1f9d-4243-8cf2-0ee86696a366)
 
 Alls good like you said